src/charm.py

```diff
@@ -399,6 +403,15 @@ def _update_hydra_endpoints_relation_data(self, event: RelationEvent) -> None:
 
     def _on_hydra_pebble_ready(self, event: WorkloadEvent) -> None:
         """Event Handler for pebble ready event."""
+        # Necessary directory for log forwarding
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Hydra container. Deferring event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Hydra container")
+            return
+        if not self._container.isdir(str(self._log_dir)):
+            self._container.make_dir(path=str(self._log_dir), make_parents=True, permissions=0o755)
```

Comment:
A log after writing the directory seems appropriate.

Reply:
added